src/Microsoft.AspNetCore.Mvc.Razor.Precompilation.Design/Internal/PrecompileRunCommand.cs

```diff
@@ -0,0 +1,228 @@
+                throw new ArgumentException($"Option {OutputPathOption.Template} does not specify a value.");
+            }
+
+            if (!ApplicationNameOption.HasValue())
+            {
+                throw new ArgumentException($"Option {ApplicationNameOption.Template} does not specify a value.");
+            }
+        }
+
+        private List<ViewCompilationInfo> ParseViews()
+        {
+            var results = new List<ViewCompilationInfo>();
+            var files = new List<RelativeFileInfo>();
+            GetRazorFiles(ServicesProvider.FileProvider, files, root: string.Empty);
+            foreach (var fileInfo in files)
+            {
+                using (var fileStream = fileInfo.FileInfo.CreateReadStream())
+                {
+                    var result = ServicesProvider.Host.GenerateCode(fileInfo.RelativePath, fileStream);
+                    results.Add(new ViewCompilationInfo(fileInfo, result));
+                }
+            }
+
+            return results;
+        }
+
+        private static void GetRazorFiles(IFileProvider fileProvider, List<RelativeFileInfo> razorFiles, string root)
+        {
+            foreach (var fileInfo in fileProvider.GetDirectoryContents(root))
+            {
+                var relativePath = Path.Combine(root, fileInfo.Name);
+                if (fileInfo.IsDirectory)
+                {
+                    GetRazorFiles(fileProvider, razorFiles, relativePath);
+                }
+                else if (fileInfo.Name.EndsWith(".cshtml", StringComparison.OrdinalIgnoreCase))
+                {
+                    razorFiles.Add(new RelativeFileInfo(fileInfo, relativePath));
+                }
+            }
+        }
+
+        private string ReadTypeInfo(CSharpCompilation compilation, SyntaxTree syntaxTree)
+        {
+            var semanticModel = compilation.GetSemanticModel(syntaxTree, ignoreAccessibility: true);
+            var classDeclarations = syntaxTree.GetRoot().DescendantNodes().OfType<ClassDeclarationSyntax>();
+            foreach (var declaration in classDeclarations)
+            {
+                var typeModel = semanticModel.GetDeclaredSymbol(declaration);
+                if (typeModel.ContainingType == null && typeModel.DeclaredAccessibility == Accessibility.Public)
+                {
+                    return GetFullName(typeModel);
+                }
+            }
+
+            return null;
+        }
+
+        private string GetFullName(INamedTypeSymbol typeModel)
+        {
+            var nameBuilder = new StringBuilder();
+
+            var containingNamespace = typeModel.ContainingNamespace;
+            while (!containingNamespace.IsGlobalNamespace)
+            {
+                nameBuilder.Insert(0, ".");
+                nameBuilder.Insert(0, containingNamespace.MetadataName);
+
+                containingNamespace = containingNamespace.ContainingNamespace;
+            }
+
+            nameBuilder.Append(typeModel.MetadataName);
```

Comment:
It doesn't, but we're only ever dealing with Razor view types and the view engine pipeline does not support instantiating generic types.